shipyard_airflow/control/airflow_connections.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2017 AT&T Intellectual Property.  All other rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import falcon
+import json
+import requests
+
+from .base import BaseResource
+
+# We need to be able to add/delete connections so that we can create/delete
+# connection endpoints that Airflow needs to connect to
+class AirflowAddConnectionResource(BaseResource):
+
+    authorized_roles = ['user']
+
+    def on_get(self, req, resp, action, conn_id, protocol, host, port):
+        # Retrieve URL
+        web_server_url = self.retrieve_config('BASE', 'WEB_SERVER')
+
+        if 'Error' in web_server_url:
+            resp.status = falcon.HTTP_400
+            resp.body = json.dumps({'Error': 'Missing Configuration File'})
+            return
+        else:
+            if action == 'add':
+                # Concatenate to form the connection URL, Remove whitespaces
+                conn_uri = "%s %s %s %s %s" % (protocol, '://', host, ':', port)
+                conn_uri = conn_uri.replace(' ', '')
+
+                # Form the request URL towards Airflow
+                req_url = '{}/admin/rest_api/api?api=connections&add=true&conn_id={}&conn_uri={}'.format(web_server_url, conn_id, conn_uri)
+            else:
+                self.return_error(resp, falcon.HTTP_400, 'Invalid Paremeters for Adding Airflow Connection')
+                return
+
+            response = requests.get(req_url).json()
+       
+            # Return output
+            if response["output"]["stderr"]:
+                resp.status = falcon.HTTP_400
```

Comment:
The idea that I have is that the codes is wrapping around CLI which is usually triggered once and returns error if it fails (be it syntax or actual failure that is reflected in stderr).  Hence I was not really looking at retry in this case.  That is the reason behind using HTTP 400.